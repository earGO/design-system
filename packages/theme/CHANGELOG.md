# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 2.0.0 (2019-08-29)

* build successfull #49 ([22545b2](https://gitlab.ursip.ru/ursip/design-system/commit/22545b2)), closes [#49](https://gitlab.ursip.ru/ursip/design-system/issues/49)
* documentations working, closing issue, moving to components and blocks development #49 ([6586776](https://gitlab.ursip.ru/ursip/design-system/commit/6586776)), closes [#49](https://gitlab.ursip.ru/ursip/design-system/issues/49)
* done rollup build #28 ([b3f59b0](https://gitlab.ursip.ru/ursip/design-system/commit/b3f59b0)), closes [#28](https://gitlab.ursip.ru/ursip/design-system/issues/28)
* got some heavy errors while mounting Tooltip and Select in storybook #45 ([3ac4051](https://gitlab.ursip.ru/ursip/design-system/commit/3ac4051)), closes [#45](https://gitlab.ursip.ru/ursip/design-system/issues/45)
* Merge branch 'modules' of gitlab.ursip.ru:ursip/design-system into modules ([af48ed9](https://gitlab.ursip.ru/ursip/design-system/commit/af48ed9))
* monorepo-structure in process #49 ([2851623](https://gitlab.ursip.ru/ursip/design-system/commit/2851623)), closes [#49](https://gitlab.ursip.ru/ursip/design-system/issues/49)
* replaced icons package with single component, working on a Collapse tests and components but have to ([b55b0d9](https://gitlab.ursip.ru/ursip/design-system/commit/b55b0d9)), closes [#45](https://gitlab.ursip.ru/ursip/design-system/issues/45)
* rollup throws and documents are not building #28 ([66262fd](https://gitlab.ursip.ru/ursip/design-system/commit/66262fd)), closes [#28](https://gitlab.ursip.ru/ursip/design-system/issues/28)
* rollup working, storybook on atoms working, storybook on cells and molecules trying to babelyfi roll ([0d42982](https://gitlab.ursip.ru/ursip/design-system/commit/0d42982)), closes [#28](https://gitlab.ursip.ru/ursip/design-system/issues/28)
* stash local before pulling homework #28 ([aa7abda](https://gitlab.ursip.ru/ursip/design-system/commit/aa7abda)), closes [#28](https://gitlab.ursip.ru/ursip/design-system/issues/28)
