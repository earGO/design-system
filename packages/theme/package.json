{
  "name": "@design-system/theme",
  "version": "1.2.0",
  "main": "build/index.es.js",
  "module": "build/index.es.js",
  "module:src": "src/index.js",
  "main:src": "src/index.js",
  "license": "MIT",
  "scripts": {
    "prepare": "rollup --config",
    "clean": "rm -rf node_modules dist"
  },
  "dependencies": {
    "color": "3.1.2"
  },
  "devDependencies": {
    "@babel/core": "^7.5.5",
    "babel-loader": "^8.0.6",
    "babel-plugin-import": "^1.12.0",
    "core-js": "3",
    "rollup": "^1.14.6",
    "rollup-plugin-babel": "^4.3.0",
    "rollup-plugin-commonjs": "^9.2.0",
    "rollup-plugin-json": "^3.1.0",
    "rollup-plugin-node-resolve": "^4.0.0",
    "rollup-plugin-peer-deps-external": "^2.2.0",
    "rollup-plugin-postcss": "^2.0.3",
    "rollup-plugin-terser": "^4.0.4",
    "rollup-plugin-url": "^2.1.0"
  }
}
