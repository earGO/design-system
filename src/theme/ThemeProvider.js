import React from 'react'
import PropTypes from 'prop-types'
import styled, { ThemeProvider as StyledThemeProvider, createGlobalStyle } from 'styled-components'
import normalize from 'styled-normalize'
import baseTheme from '.'
console.log(normalize)

const GlobalStyle = createGlobalStyle`
html,
body {
  height: 100%;
}

${normalize};
`

export const Base = styled.div`
  font-family: ${props => props.theme.font.main};
  line-height: 1.4;
  height: 100%;

  * {
    box-sizing: border-box;
  }
`

const ThemeProvider = ({ customBreakpoints, ...props }) => {
  const breakpoints = customBreakpoints || baseTheme.breakpoints
  const theme = {
    ...baseTheme,
    ...props.theme,
    breakpoints,
  }

  return (
    <React.Fragment>
      <GlobalStyle />
      <StyledThemeProvider theme={theme}>
        <Base {...props} />
      </StyledThemeProvider>
    </React.Fragment>
  )
}

ThemeProvider.propTypes = {
  /**
   * Массив значений в пикселях для кастмизации дефолтных брейкпоинтов
   */
  customBreakpoints: PropTypes.arrayOf(PropTypes.number),
}

export default ThemeProvider
