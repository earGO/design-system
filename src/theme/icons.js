import { library } from '@fortawesome/fontawesome-svg-core'
import { faBeer, faCog, faSpinner, faCheck, faChevronDown, faQuestionCircle } from '@fortawesome/free-solid-svg-icons'
// import { faQuestionCircle } from '@fortawesome/free-regular-svg-icons'

library.add(faBeer, faCog, faSpinner, faCheck, faChevronDown, faQuestionCircle)

export default library
