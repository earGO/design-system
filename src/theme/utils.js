import Color from 'color'

export function darken(color, percent) {
  return Color(color)
    .darken(percent)
    .hex()
}

export function lighten(color, percent) {
  return Color(color)
    .lighten(percent)
    .hex()
}

export function makeShades(colorName, colorValue) {
  const shades = {
    dark: color => darken(color, 0.7),
    semiDark: color => darken(color, 0.35),
    light: color => lighten(color, 0.7),
    semiLight: color => lighten(color, 0.35),
  }

  return Object.keys(shades).reduce(
    (acc, shade) => ({
      ...acc,
      [`${shade}${colorName}`]: shades[shade](colorValue),
    }),
    {},
  )
}

export const addAliases = (arr, aliases) =>
  aliases.forEach((key, i) =>
    Object.defineProperty(arr, key, {
      enumerable: false,
      get() {
        return this[i]
      },
    }),
  )
