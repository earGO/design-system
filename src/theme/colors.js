import csscolors from 'css-color-names'
import { lighten, darken } from './utils'

const black = '#3a3a3a'
const white = '#ffffff'

// greys
const lightGrey = '#f5f5f5'
const semiLightGrey = '#ecebeb'
const grey = '#b5b5b5'

const red = '#ef5350'
const green = '#66bb6a'
// blues
const lightBlue = '#0091ea'
const blue = '#1e88e5'
const darkBlue = darken(blue, 0.75)

const palette = {
  ...csscolors,
  black,
  white,
  lightGrey,
  semiLightGrey,
  grey,
  lightBlue,
  blue,
  darkBlue,
  red,
  green,
}

const components = {
  // Input element
  input: {
    hover: palette.black,
    disabled: palette.grey,
    error: palette.red,
    success: palette.green,
    focus: palette.lightBlue,
  },

  // Checkbox
  checkbox: {
    checked: palette.lightBlue,
    unchecked: palette.lightGrey,
    disabled: palette.grey,
  },

  //Tabs
  tabs: {
    active: palette.blue,
    hover: palette.lightBlue,
  },
  //Radio
  radio: {
    checked: palette.lightBlue,
    unchecked: palette.lightGrey,
    disabled: palette.grey,
  },
}

const system = {
  // Main color
  primary: palette.blue,

  // States
  info: palette.lightBlue,
  success: palette.green,
  warning: palette.orange,
  error: palette.red,

  // Misc
  text: palette.darkBlue,
  border: palette.semiLightGrey,

  // Disabled
  disabled: palette.grey,
  // Highlight? placeholder for now.
  highlight: palette.lightGrey,
  highlightHover: palette.lightBlue,
  // Scrollbar
  scrollbar: palette.grey,
  // border: palette.semiLightGrey,
}

export default {
  system,
  palette,
  ...palette,
  ...system,
  ...components,
}
