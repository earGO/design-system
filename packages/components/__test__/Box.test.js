import React from 'react'
import renderer from 'react-test-renderer'
import Box from '../src/Box'
import theme, {ThemeProvider} from '@design-system/theme'

describe('Box', () => {
  test('renders without props', () => {
    const json = renderer.create(<Box />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('width prop sets width', () => {
    const json = renderer.create(<Box width={1 / 2} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('width', '50%')
  })

  test('m prop sets margin', () => {
    const json = renderer.create(<Box m={2} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('margin', theme.space[2] + 'px')
  })

  test('p prop sets padding', () => {
    const json = renderer.create(<Box p={2} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('padding', theme.space[2] + 'px')
  })

  test('color prop sets color', () => {
    const json = renderer.create(<Box color={theme.colors.blue} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('color', theme.colors.blue)
  })
  test('bg prop sets background color', () => {
    const json = renderer.create(<Box bg={theme.colors.green} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
  })
  test('recieves theme colors', () => {
    const json = renderer
      .create(
        <Box width={1 / 2} bg="primary" color="white" theme={theme}>
          Hello Box
        </Box>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.primary)
  })
  test('align prop triggers warning', () => {
    console.error = jest.fn()
    const json = renderer.create(<Box align="center" />).toJSON()
    expect(
      console.error.mock.calls
        .toString()
        .indexOf(
          'The Box `align` prop will deprecated. Please use Text instead.'
        ) !== -1
    )
    console.error.mockRestore()
  })
})
