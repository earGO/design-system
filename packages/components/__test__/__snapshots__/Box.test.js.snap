// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box bg prop sets background color 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  background-color: #66bb6a;
}

<div
  className="c0"
/>
`;

exports[`Box color prop sets color 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  color: #0091ea;
}

<div
  className="c0"
  color="#0091ea"
/>
`;

exports[`Box m prop sets margin 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  margin: 8px;
}

<div
  className="c0"
/>
`;

exports[`Box p prop sets padding 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  padding: 8px;
}

<div
  className="c0"
/>
`;

exports[`Box renders 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`Box width prop sets width 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  width: 50%;
}

<div
  className="c0"
  width={0.5}
/>
`;
